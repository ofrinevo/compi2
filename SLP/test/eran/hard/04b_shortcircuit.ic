	
class A {
	static boolean true_print() {
	 Library.println("Should not be printed");
	 return true;
	}
} 

class Check {
	static void main (string[] args) { 
		
		Library.println("short circuit");
		boolean b = true;
		if (b || A.true_print()) {
		 Library.println("This is ok");
		}
 	}
}
