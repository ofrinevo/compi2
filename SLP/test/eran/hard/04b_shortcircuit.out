short circuit
This is ok
